homeassistant/components/ecobee/climate.py

```diff
@@ -11,8 +11,6 @@
     ATTR_TARGET_TEMP_LOW,
     FAN_AUTO,
     FAN_ON,
-    PRESET_AWAY,
-    PRESET_NONE,
```

Comment:
Is there always a preset as `"none"` is removed?

Reply:
This comment and related code block should help explain:
https://github.com/home-assistant/core/pull/108636/files#r1461393304

In short, this is both no longer accessible (due to the newly-added validation), *and* there's a different, documented way to resume program that actually makes sense.

To answer your question, I don't believe there always needs to be a preset active. You can see the previous logic (which I mostly left intact) in `preset_mode()` - it never returned `PRESET_NONE`.

Reply:
If there isn't a preset set it should return `PRESET_NONE`.
I noticed in the current implementation it seems somehow to be used only in setting the preset but not as an option in `preset_modes`.
Anyhow this should be fixed if actually now there isn't always a preset.

Reply:
Conceptually that might make sense, but from what I understand this may be the current behavior (it's a bit hard to understand the full set of logic/states that are possible in this component):
```
# Any hold not based on a climate is a temp hold
return PRESET_TEMPERATURE
```

This PR doesn't change that, nor does it change whether or not there is always a preset. That feels outside the scope of this, the intention being to simply add one new preset - indefinite away. While I was in the area, I cleaned up a bunch of old presets that are completely inaccessible. No other behavior has changed.